src/lib/utils/memory_usage.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+
+namespace opossum {
+
+class MemoryUsage final {
+ public:
+  explicit MemoryUsage(const size_t bytes);
+
+  void print(std::ostream& stream) const;
```

Comment:
Would it make sense to overload the `<<` operator instead?

Reply:
Yes. But it's established practice in hyrise right now to provide a print function. We should refactor all of them to be operator<< at some point, imo, but not as part of this PR. Also, because I want to use this https://github.com/fmtlib/fmt